flake.nix

```diff
@@ -28,17 +28,31 @@
     # Run checks: nix flake check -L
     checks = forAllSystems (system: let
       pkgs = nixpkgs.legacyPackages.${system};
+      lib = nixpkgs.lib;
       nixosTests = import ./tests {
         inherit pkgs;
         makeTest = import (pkgs.path + "/nixos/tests/make-test-python.nix");
         eval-config = import (pkgs.path + "/nixos/lib/eval-config.nix");
       };
+      documentation = (import (pkgs.path + "/nixos/doc/manual") rec {
```

Comment:
There is actual a proper api doc module documentation now. I had some wip branch here: https://github.com/nix-community/disko/pull/117
But also same issue because of those special subType the documentation cannot discover all options in disko.
To solve this issue one would need to add the equivalent of https://github.com/NixOS/nixpkgs/blob/master/lib/types.nix#L554-L555

Reply:
Yeah I'll close this one in favor of #117, since that one is clearly a better approach